### PR TITLE
[ioctl] fix log entries created from user input

### PR DESCRIPTION
**Pull request by huof6829**
# Description

fix code scan#7~#10 https://github.com/iotexproject/iotex-core/security/code-scanning/10

Fixes https://github.com/iotexproject/iotex-core/issues/3526

## Type of change
Please delete options that are not relevant.
- [] Bug fix (non-breaking change which fixes an issue)
- [] New feature (non-breaking change which adds functionality)
- [] Code refactor or improvement
- [] Breaking change (fix or feature that would cause a new or changed behavior of existing functionality)
- [] This change requires a documentation update

# How Has This Been Tested?
Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration
- [] make test
- [] fullsync
- [] Other test (please specify)

**Test Configuration**:
- Firmware version:
- Hardware:
- Toolchain:
- SDK:

# Checklist:
- [] My code follows the style guidelines of this project
- [] I have performed a self-review of my code
- [] I have commented my code, particularly in hard-to-understand areas
- [] I have made corresponding changes to the documentation
- [] My changes generate no new warnings
- [] I have added tests that prove my fix is effective or that my feature works
- [] New and existing unit tests pass locally with my changes
- [] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: ioctl/cmd/contract/contractshare.go [hunk 1]
```diff
@@ -233,21 +234,26 @@ func share(args []string) error {
...
 				if err != nil {
 					log.Println("set file failed: ", err)
+					break
 				}
 				if err := conn.WriteJSON(&response); err != nil {
 					log.Println("send set response: ", err)
 					break
 				}
-				log.Println("set: " + _givenPath + "/" + setPath)
+				log.Printf("set: %s/%s\n", _givenPath, easpcapeString(setPath))
```
Comment (Liuhaai) [inline ioctl/cmd/contract/contractshare.go:243]: why is easpcapeString added here

Comment (huof6829) [reply ioctl/cmd/contract/contractshare.go:243]: user can make many new log entries, this is unsafe.   https://github.com/iotexproject/iotex-core/security/code-scanning/10

Comment (Liuhaai) [reply ioctl/cmd/contract/contractshare.go:243]: good. Could you reproduce this issue locally?

File: ioctl/cmd/contract/contractshare.go [hunk 2]
```diff
@@ -233,21 +234,26 @@ func share(args []string) error {
...
 		}
 	})
 	log.Fatal(http.ListenAndServe(*_addr, nil))
 
 	return nil
+}
 
+func easpcapeString(str string) string {
+	escaped := strings.Replace(str, "\n", "", -1)
+	return strings.Replace(escaped, "\r", "", -1)
```
Comment (CoderZhi) [inline ioctl/cmd/contract/contractshare.go:258]: try `return strings.Replace(escaped, "[\r\n]", "", -1)

**Review by CoderZhi — APPROVED** [review]

**Review by Liuhaai — APPROVED** [review]

**Pull request merged**